be2-scala/src/test/scala/ch/epfl/pop/model/objects/ChannelSuite.scala

```diff
@@ -73,8 +73,26 @@ class ChannelSuite extends FunSuite with Matchers {
 
     noException shouldBe thrownBy(channel)
     channel.decodeSubChannel.get should equal(expected)
+  }
+
+  test("Real LaoId extraction from the middle channel test") {
+    val laoId = "mEKXWFCMwb";
+    def channel = Channel(Channel.ROOT_CHANNEL_PREFIX + Base64Data.encode(laoId) + Channel.SEPARATOR + Base64Data.encode("social"))
+    val expected = laoId.getBytes()
 
+    noException shouldBe thrownBy(channel)
+    channel.decodeChannelLaoId.get should equal(expected)
   }
+
+  test("Real LaoId extraction fails with wrong channel structure (not base64Data)") {
+    val laoId = "base64_lao_id"; // Not encoded in BASE64
```

Comment:
The beauty of fake variables, is that the comment can be the variable 😄 
```suggestion
    val laoId = "not_a_base64_lao_id";
```

Reply:
fair enough thank you!